### PR TITLE
add more stats: connector count, connector/config index status; fix model count bug

### PR DESCRIPTION
**Pull request by ylwu-amzn**
### Description
Add more stats:
- connector count
- connector index status
- config index status

Fix model count bug: local model may have multiple chunks, the model count number should not include model chunk.

Also fine tune some error message
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: plugin/src/main/java/org/opensearch/ml/model/MLModelGroupManager.java [hunk 1]
```diff
@@ -77,7 +77,7 @@ public void createModelGroup(MLRegisterModelGroupInput input, ActionListener<Str
                                     new IllegalArgumentException(
                                         "The name you provided is already being used by another model with ID: "
                                             + id
-                                            + ". Please provide a different name"
+                                            + ". Please provide a different name or add \"model_group_id\": \"lMPmr4kB4eSCtCCDmCDm\" to request body"
```
Comment (jackiehanyang) [inline plugin/src/main/java/org/opensearch/ml/model/MLModelGroupManager.java:80]: why are we adding a hard coded model id in the error message?

File: ml-algorithms/src/main/java/org/opensearch/ml/engine/encryptor/EncryptorImpl.java [hunk 2]
```diff
@@ -34,6 +34,7 @@
 @Log4j2
 public class EncryptorImpl implements Encryptor {
 
+    public static final String MASTER_KEY_NOT_READY_ERROR = "ML encryption master key not initialized yet. Please retry after 10 seconds.";
```
Comment (dhrubo-os) [inline ml-algorithms/src/main/java/org/opensearch/ml/engine/encryptor/EncryptorImpl.java:37]: How about: `The ML encryption master key has not been initialized yet. Please retry after waiting for 10 seconds.`

File: plugin/src/main/java/org/opensearch/ml/model/MLModelGroupManager.java (hunk 1 above)
Comment (dhrubo-os) [inline plugin/src/main/java/org/opensearch/ml/model/MLModelGroupManager.java:80]: Should we update to `Please provide a different group name`. We can see people started having confusion: https://github.com/opensearch-project/ml-commons/issues/1179

about: `\"model_group_id\": \"lMPmr4kB4eSCtCCDmCDm\"` :  Is this any static group id we are adding?

Comment (ylwu-amzn) [reply plugin/src/main/java/org/opensearch/ml/model/MLModelGroupManager.java:80]: Oh, good catch, I need to clean up the code first.

Comment (ylwu-amzn) [reply plugin/src/main/java/org/opensearch/ml/model/MLModelGroupManager.java]: Fixed hard coded model id .
For `Please provide a different model group name` , that looks also confusing as user just provide model name in the request body, not model group name,

```
POST _plugins/_ml/models/_upload
{
  "name": "huggingface/sentence-transformers/all-MiniLM-L12-v2",
  "version": "1.0.1",
  "model_format": "TORCH_SCRIPT"
}
```

**Review by dhrubo-os — APPROVED** [review]

**Review by Zhangxunmt — APPROVED** [review]

**Review by Zhangxunmt — COMMENTED** [review]
We may add more granularity stats like connector/model count for sageMaker, etc, if necessary in the future.

**Pull request merged**